### PR TITLE
fix(cli): use CI's commit environment variables

### PR DESCRIPTION
**Pull request by YoussefRaafatNasry**
The `gitProviderSha` method was using `git rev-list` to get the SHA of the triggering commit. This PR aims to use the following environment variables when running on CI.

| Provider | Environment Variable |
| :--- | :--- |
| GitHub | [`GITHUB_SHA`](https://docs.github.com/en/actions/learn-github-actions/variables#:~:text=github.com.-,GITHUB_SHA,-The%20commit%20SHA) |
| Codemagic | [`CM_COMMIT`](https://docs.codemagic.io/yaml-basic-configuration/environment-variables/#:~:text=from%20custom%20source-,CM_COMMIT,-Commit%20hash%20that) |

### CODE REVIEW

File: packages/widgetbook_cli/bin/commands/publish.dart [hunk 1]
```diff
@@ -210,34 +213,16 @@ class PublishCommand extends WidgetbookCommand {
   }
 
   @visibleForTesting
-  Future<String?> gitProviderSha({
-    required GitDir gitDir,
-  }) async {
-    final commits = await gitDir.commits();
```
Comment (YoussefRaafatNasry) [inline packages/widgetbook_cli/bin/commands/publish.dart:216]: The `commits()` method is no longer used in anywhere of our codebase, and can be safely removed with it's internal implementation methods and tests. This will be done in a separate PR to allow reverting in case it's needed.

File: .github/workflows/deploy-widgetbook-feat.yml [hunk 2]
```diff
@@ -30,7 +30,7 @@ jobs:
         run: flutter build web -t lib/app.widgetbook.dart
 
       - name: Install Widgetbook CLI
-        run: dart pub global activate widgetbook_cli 3.0.0-beta.22
+        run: dart pub global activate --source path ../packages/widgetbook_cli
```
Comment (YoussefRaafatNasry) [inline .github/workflows/deploy-widgetbook-feat.yml:33]: Should this change be done in a separate PR for more visibility?

**Review by jenshor — APPROVED** [review]
LGTM 🎉 Great job, Youssef 🚀

File: packages/widgetbook_cli/bin/commands/publish.dart (hunk 1 above)
Comment (jenshor) [reply packages/widgetbook_cli/bin/commands/publish.dart:216]: Feel free to remove the method.
